posts/2023-08-24-Rust-1.72.0.md

```diff
@@ -0,0 +1,60 @@
+  - The second checks for calls to `std::str::from_utf8` and `std::str::from_utf8_mut` with an invalid UTF-8 literal, which will always return an error. This lint is a warning by default.
+
+* [`clippy::cmp_nan`](https://rust-lang.github.io/rust-clippy/rust-1.71.0/index.html#cmp_nan) to [`invalid_nan_comparisons`](https://doc.rust-lang.org/1.72.0/rustc/lints/listing/warn-by-default.html#invalid-nan-comparisons) (warn)
+  - This checks for comparisons with `f32::NAN` or `f64::NAN` as one of the operands. NaN does not compare meaningfully to anything – not even itself – so those comparisons are always false. This lint is a warning by default, and will suggest calling the `is_nan()` method instead.
+
+* [`clippy::cast_ref_to_mut`](https://rust-lang.github.io/rust-clippy/rust-1.71.0/index.html#cast_ref_to_mut) to [`invalid_reference_casting`](https://doc.rust-lang.org/1.72.0/rustc/lints/listing/allowed-by-default.html#invalid-reference-casting) (allow)
+  - This checks for casts of `&T` to `&mut T` without using interior mutability, which is immediate undefined behavior, even if the reference is unused. This lint is currently allowed by default, but it is planned to be denied by default in 1.73 after implementation improvements.
+
+### Stabilized APIs
+
+- [`impl<T: Send> Sync for mpsc::Sender<T>`](https://doc.rust-lang.org/stable/std/sync/mpsc/struct.Sender.html#impl-Sync-for-Sender%3CT%3E)
+- [`impl TryFrom<&OsStr> for &str`](https://doc.rust-lang.org/stable/std/primitive.str.html#impl-TryFrom%3C%26'a+OsStr%3E-for-%26'a+str)
+- [`String::leak`](https://doc.rust-lang.org/stable/alloc/string/struct.String.html#method.leak)
+
+These APIs are now stable in const contexts:
+
+- [`CStr::from_bytes_with_nul`](https://doc.rust-lang.org/stable/std/ffi/struct.CStr.html#method.from_bytes_with_nul)
+- [`CStr::to_bytes`](https://doc.rust-lang.org/stable/std/ffi/struct.CStr.html#method.to_bytes)
+- [`CStr::to_bytes_with_nul`](https://doc.rust-lang.org/stable/std/ffi/struct.CStr.html#method.to_bytes_with_nul)
+- [`CStr::to_str`](https://doc.rust-lang.org/stable/std/ffi/struct.CStr.html#method.to_str)
+
+### Other changes
+
+Check out everything that changed in [Rust](https://github.com/rust-lang/rust/releases/tag/1.72.0), [Cargo](https://github.com/rust-lang/cargo/blob/master/CHANGELOG.md#cargo-172-2023-08-24), and [Clippy](https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#rust-172).
+
+In a future release we're planning to increase the minimum supported Windows version to 10. The accepted proposal in compiler [MCP 651](https://github.com/rust-lang/compiler-team/issues/651) is that Rust 1.75 will be the last to officially support Windows 7, 8, and 8.1. When Rust 1.76 is released in February 2024, only Windows 10 and later will be supported as tier-1 targets.
```

Comment:
This feels a bit tucked away. Maybe extract it from "Other changes" and give it its own header?

Reply:
We could -- I was just repeating the way we pre-announced a Linux bump in 1.61:
https://blog.rust-lang.org/2022/05/19/Rust-1.61.0.html#other-changes

How about a heading like, "Future Windows compatibility"?

Reply:
Not opposed to this but we should probably pin down what the requirements are for updating a target. E.g. if dropping old windows needs only an MCP and mention in a release post, it's a bit odd that dropping old macOS needs an MCP, multi-team FCP, and dedicated blog post (especially when old macOS is far more obscure than old windows).

Reply:
Yeah, I'm just surprised to see this mentioned without doing that for Windows.

Reply:
It was requested [here](https://rust-lang.zulipchat.com/#narrow/stream/241545-t-release/topic/1.2E72.2E0/near/386580207) -- if we feel this is premature, we can wait for a later release. Again looking at the Linux case, the initial mention in release notes was 2 months after FCP was _proposed_, but that was blocked on having a dedicated blog post, which was posted 3 months later.

Reply:
^cc @ChrisDenton

Reply:
For Windows XP we didn't have all of that. But then nobody knew what the process was at that time so I just jumped on that when someone suggested it. I dunno. There's no official process so it feels like it's always different every time.

Reply:
I really do want to give people as much notice as possible. Maybe it could be hedged? I mean, "planned" doesn't necessarily mean it's definite.

Reply:
I think we should keep this notice. I don't think it should be taken as implying this is *enough* or not enough -- this is not a post stating what the requirements are for future changes (or even this change). If we push out the actual switch I doubt people will be unhappy, all this really commits us to (and even that lightly) is a *minimum* delay until 1.75.